### DISCUSSION TITLE
Trouble syncing a new desktop client

### DISCUSSION THREAD

**Discussion by dousha** [TriliumNext/Trilium#2866] [Q&A]
I'm setting up a new desktop client on a new computer. However, I noticed that the initial syncing process takes forever to finish. The outstanding change count is stuck on 1000+ with no network activity.

Initially, I suspected that Nginx was closing the connection prematurely since the connection stalled after about 3 minutes after the initial handshake. But there is nothing unusual in the access log and the error log. Setting `proxy_read_timeout`, `proxy_send_timeout` both to `1d` has no effect. The connection still stalls after about 3 minutes.

Firewall configurations seemed normal. There is no rule on closing a long-running connection. (I don't think `firewalld` has such an ability.)

Examining Trilium's server log yielded no result. It seems that the client could talk with the server but the connection just stalls for no apparent reason.

I also tested if my VPS provider would interfere with large file transmissions. Downloading a 100MiB test file via SSH and `serve` directly on port 80/443 all seemed fine.

Is there any pointer to further diagnose this issue?

**Comment by zadam** [comment]
Hi, could you share your client log? Either here or to zadam.apps@gmail.com. Thanks.

**Reply by dousha** [reply]
Sorry for the late reply. My computer died for some bizarre reasons and I had just fixed it now.

After tallying the client log, I found that the client reported the synchronization process had failed. Interestingly, the failure was reported way earlier than the network stall so I didn't catch it in the first place as it has scrolled past the terminal screen. Client-side log is [available here](https://paste.ubuntu.com/p/bRcK7Fnd74/).

It seems that Trilium would only wait for 120 seconds for the synchronization process. This might be an issue if the note database is large and/or the user has a slow connection to the server. Esp. self-hosting servers with a rather limited bandwidth usually start at only 1Mbit/s.

**Reply by zadam** [reply]
Hi, 120 seconds is not a time limit for the whole sync, but for individual requests and this should be enough. You can configure this limit in the options:

![image](https://user-images.githubusercontent.com/617641/170864987-7e0bbe77-b4b8-4982-882e-c0666d8c1eb5.png)

However it's quite possible/probable that the problem is not the timeout, but some other network issue.
